components/utils.ts

```diff
@@ -189,13 +189,35 @@ export const SNIPER_LEVEL_INFO: number[] = [
     38000000, 47000000, 58000000, 70000000,
 ]
 
+/**
+ * Get the number of xp needed to reach a level in sniper missions.
+ * @param level The level in question.
+ * @returns The xp, as a number.
+ */
+export function xpRequiredForSniperLevel(level: number): number {
+    return SNIPER_LEVEL_INFO[level - 1]
+}
+
 /**
  * Clamps the given value between a minimum and maximum value
  */
 export function clampValue(value: number, min: number, max: number) {
     return Math.max(min, Math.min(value, max))
 }
 
+/**
+ * Returns whether a location is a sniper location. Works for both parent and child locations.
+ * @param location The location ID string.
+ * @returns A boolean denoting the result.
+ */
+export function isSniperLocation(location: string): boolean {
+    return (
+        location.includes("AUSTRIA") ||
+        location.includes("SALTY") ||
+        location.includes("CAGED")
+    )
+}
+
```

Comment:
Question: `Contract["Metadata"]["Type"] == "sniper"`?

Reply:
But this function is for the location as a string? Do you mean I should consult `missionsInLocation` and then `resolveContract`? That's a long detour...

Reply:
Ah, I thought when you passed through the location it was from a contract. My apologies.